### PR TITLE
Created stewardship activities documentation

### PR DESCRIPTION
**Pull request by florisdobber**

### CODE REVIEW

File: docs/sftt/sftt-api-spec.md [hunk 1]
```diff
@@ -1681,3 +1681,92 @@ If the site id specified is invalid.
...
+`POST api/v1/protected/sites/:site_id/record_stewardship`
+
+Records a stewardship activity for the given site. Indicate `True` if the activity (watered, mulched, etc.) was completed, else `False`.
+
+#### Request Body
+
+```json
+{
+  "date": TIMESTAMP,
+  "duration": INT | NULL,
```
Comment (liammoyn) [inline docs/sftt/sftt-api-spec.md:1696]: I think we should indicate here the units we're using for duration. We can do that be changing the name of the key to be minutesDuration, if you're super opposed to that then we can just have a note here that specifies that duration should be counted as minutes

File: docs/sftt/sftt-api-spec.md [hunk 2]
```diff
@@ -1681,3 +1681,92 @@ If the site id specified is invalid.
...
+
+`POST api/v1/protected/sites/:site_id/record_stewardship`
+
+Records a stewardship activity for the given site. Indicate `True` if the activity (watered, mulched, etc.) was completed, else `False`.
+
+#### Request Body
+
+```json
+{
+  "date": TIMESTAMP,
```
Comment (liammoyn) [inline docs/sftt/sftt-api-spec.md:1695]: Should this be a TIMESTAMP or a date? I'm assuming this field is something the user enters, will they also be adding a time when they performed it?

File: docs/sftt/sftt-api-spec.md [hunk 3]
```diff
@@ -1681,3 +1681,92 @@ If the site id specified is invalid.
...
+
+##### `401 UNAUTHORIZED`
+
+If the calling user is not an admin or the user listed on the activity.
+
+### Get Stewardship Activities By Site
+
+`GET api/v1/protected/sites/:site_id/stewardship_activities`
+
+Returns all the recorded stewardship activities for the indicated site.
```
Comment (liammoyn) [inline docs/sftt/sftt-api-spec.md:1742]: Will this be sorted? If so specify

File: docs/sftt/sftt-api-spec.md (hunk 1 above)
Comment (SofieDunt) [inline docs/sftt/sftt-api-spec.md:1696]: Should we require users to enter a duration? It doesn't seem like a lot to ask and might be helpful information

File: docs/sftt/sftt-api-spec.md (hunk 2 above)
Comment (SofieDunt) [reply docs/sftt/sftt-api-spec.md:1695]: Date might be sufficient and more convenient - I believe NY doesn't ask for time

File: docs/sftt/sftt-api-spec.md [hunk 4]
```diff
@@ -1681,3 +1681,92 @@ If the site id specified is invalid.
...
+#### Request Body
+
+```json
+{
+  "date": TIMESTAMP,
+  "duration": INT | NULL,
+  "watered": BOOLEAN,
+  "mulched": BOOLEAN,
+  "cleaned": BOOLEAN,
+  "weeded": BOOLEAN
```
Comment (SofieDunt) [inline docs/sftt/sftt-api-spec.md:1700]: 💭 We should probably check that these aren't all false. Could be done on the frontend

File: docs/sftt/sftt-api-spec.md [hunk 5]
```diff
@@ -1681,3 +1681,92 @@ If the site id specified is invalid.
...
+##### `200 OK`
+
+```json
+{
+  "activities": [
+    {
+      "id": INT,
+      "user_id": INT,
+      "date": TIMESTAMP,
+      "duration": INT,
```
Comment (SofieDunt) [inline docs/sftt/sftt-api-spec.md:1759]: ❓ Would this be `| NULL`?

File: docs/sftt/sftt-api-spec.md (hunk 1 above)
Comment (florisdobber) [reply docs/sftt/sftt-api-spec.md:1696]: Noted in duration is time in minutes

File: docs/sftt/sftt-api-spec.md (hunk 2 above)
Comment (florisdobber) [reply docs/sftt/sftt-api-spec.md:1695]: Good point, changed to DATE

File: docs/sftt/sftt-api-spec.md (hunk 1 above)
Comment (florisdobber) [reply docs/sftt/sftt-api-spec.md:1696]: SFTT specifically asked we don't require it. If you want to require it it's an easy fix, we can ask tomorrow.

File: docs/sftt/sftt-api-spec.md (hunk 4 above)
Comment (florisdobber) [reply docs/sftt/sftt-api-spec.md:1700]: Do we want to do this on the back end too? Feels like it should be done on the front end but I'm not sure if it's good to also require on the back end. @liammoyn

File: docs/sftt/sftt-api-spec.md [hunk 6]
```diff
@@ -1681,3 +1681,92 @@ If the site id specified is invalid.
...
+
+`POST api/v1/protected/sites/:site_id/record_stewardship`
+
+Records a stewardship activity for the given site. Duration is the time taken to perform the activity in minutes. Indicate `True` if the activity (watered, mulched, etc.) was completed, else `False`.
+
+#### Request Body
+
+```json
+{
+  "date": DATE,
```
Comment (SofieDunt) [inline docs/sftt/sftt-api-spec.md:1695]: Would we want to change this to `performed_on`/describe what date this is too?

File: docs/sftt/sftt-api-spec.md (hunk 4 above)
Comment (liammoyn) [reply docs/sftt/sftt-api-spec.md:1700]: Yeah we should do it on both. Lets specify it here so we don't forget during implementation, could throw a 400 if all are false

File: docs/sftt/sftt-api-spec.md (hunk 6 above)
Comment (liammoyn) [reply docs/sftt/sftt-api-spec.md:1695]: Probably yeah

Comment (florisdobber) [reply docs/sftt/sftt-api-spec.md:1695]: Opted for description

File: docs/sftt/sftt-api-spec.md (hunk 4 above)
Comment (florisdobber) [reply docs/sftt/sftt-api-spec.md:1700]: Fixed

File: docs/sftt/sftt-api-spec.md [hunk 7]
```diff
@@ -1681,3 +1681,92 @@ If the site id specified is invalid.
 ##### `401 UNAUTHORIZED`
 
 If the calling user is not an admin.
+
+### Record Stewardship Activity
+
+`POST api/v1/protected/sites/:site_id/record_stewardship`
+
+Records a stewardship activity for the given site. Date is the day on which the activity was performed, which can be in the past if the user is adding a past activity. Duration is the time taken to perform the activity in minutes. Indicate `True` if the activity (watered, mulched, etc.) was completed, else `False`.  At least one activity must be `True`.
```
Comment (SofieDunt) [inline docs/sftt/sftt-api-spec.md:1689]: 💯

**Review by SofieDunt — APPROVED** [review]
Nice!

**Review by liammoyn — APPROVED** [review]

**Comment by florisdobber** [comment]
GitHub messed up the merge so I ended up copy pasting the changes and pushing directly to master.

**Pull request closed without merging**